### ISSUE TITLE
Website chart display error after v1.34.0 update

### ISSUE THREAD

**Issue by hans362** [umami-software/umami#1320]
Website chart display error after v1.34.0 update
![](https://user-images.githubusercontent.com/32535810/180214046-96f1ab70-6aa6-4e09-99c1-e0c58d347981.png)
![](https://user-images.githubusercontent.com/32535810/180214686-fdbcbb5a-2baf-40a8-93e1-617d0cfa741c.png)
It seems that there's an issue with the pageviews api.

**Comment by njfdev** [comment]
Weird, I just updated and I don’t have this issue. Could you please provide more information on you setup.

**Comment by hans362** [comment]
Operating system: Ubuntu 18.04.6
Node.js version: 16.16.0
MariaDB version: 10.3.35
Everything worked fine in version 1.33.1 until updated to 1.34.0 with the following command:
```
git pull
yarn install
yarn build
yarn update-db
```

**Comment by njfdev** [comment]
Can you please check if your database has the same data?

**Comment by njfdev** [comment]
This could possibly be a MySQL database issue because I use PostgreSQL through Supabase.

**Comment by hans362** [comment]
> Can you please check if your database has the same data?

The data in database seem correct. The numeric data displayed on the dashboard is also correct. Only the bar chart is wrong. Quite weird.

**Comment by Akiracr** [comment]
I've updated and have the same error with the chart, i'm using Maria DB 10.3.34-MariaDB-0ubuntu0.20.04.1.
![Selección_326](https://user-images.githubusercontent.com/283008/180235406-b3d9b04b-9d8b-441a-8025-4c7a9aea48bf.png)

**Comment by mikecao** [comment]
Seems to be MySQL specific, investigating.

**Comment by Akiracr** [comment]
The problem seems to be in the query of getPageviewStats, i've moved getDateStringQuery into the main query and the problem disappear.

Like this:

`  return rawQuery(
    `
    select
      g.t as t,
      g.y as y
    from
      (select ${getDateStringQuery(getDateQuery('pageview.created_at', unit, '+06:00'), unit)} t,
        count(${count}) y
      from pageview
        ${joinSession}
      where pageview.website_id=$1
      and pageview.created_at between $2 and $3
      ${pageviewQuery}
      ${sessionQuery}
      group by 1) g
    order by 1
    `,
    params,
  );`

![Selección_327](https://user-images.githubusercontent.com/283008/180272075-8aaaaf8e-2ce7-49b9-824a-7defc5cb7241.png)

**Comment by mikecao** [comment]
I just pushed a fix to the `dev` branch. Can you try pulling from there and seeing if it works?

**Comment by Akiracr** [comment]
I've fecthed dev and rebuild but the endpoint api/website/1/pageviews returns an error 500
![Selección_328](https://user-images.githubusercontent.com/283008/180328563-31b7a4cf-df34-445d-8e6c-2a7dc58fd8bf.png)

Note: sorry i'm not familiarized with nodejs and does not know how to get a log to review the error. If you have a link that explain how to do it i can review it and implemented.

**Comment by hans362** [comment]
Also getting a 500 error here after pulling the dev branch
![](https://user-images.githubusercontent.com/32535810/180336082-b9740d71-fec9-428a-8e73-47a49e5f5e31.png)
![](https://user-images.githubusercontent.com/32535810/180336089-8c272fe5-1a07-431e-a2e4-70f55d0296db.png)

**Comment by DarrenOfficial** [comment]
Happened to me as well
Browser: Safari iPad OS 16
System:
 - Kernel:  4.18.0-372.9.1.el8.x86_64, Rocky Linux
 - Node: 16.16.0
 - Yarn: 1.22.18

![3D21484D-089B-4030-ADE1-69A7F33EE63C](https://user-images.githubusercontent.com/55406094/180341451-b822533a-671f-43b0-9197-73156f12bbe8.jpeg)

**Comment by mskian** [comment]
I am also having this issue

mysql 8

**Comment by mikecao** [comment]
Can someone post the actual 500 error?

**Comment by hans362** [comment]
```
Jul 22 12:15:35 S4 env[29693]: PrismaClientKnownRequestError: Raw query failed. Code: `1582`. Message: `Incorrect parameter count in the call to native function 'convert_tz'`
Jul 22 12:15:35 S4 env[29693]:     at RequestHandler.handleRequestError (/vms/umami/node_modules/@prisma/client/runtime/index.js:28658:13)
Jul 22 12:15:35 S4 env[29693]:     at RequestHandler.request (/vms/umami/node_modules/@prisma/client/runtime/index.js:28640:12)
Jul 22 12:15:35 S4 env[29693]:     at async consumer (/vms/umami/node_modules/@prisma/client/runtime/index.js:29618:18)
Jul 22 12:15:35 S4 env[29693]:     at async Proxy._request (/vms/umami/node_modules/@prisma/client/runtime/index.js:29639:16)
Jul 22 12:15:35 S4 env[29693]:     at async __WEBPACK_DEFAULT_EXPORT__ (/vms/umami/.next/server/pages/api/website/[id]/events.js:146:24)
Jul 22 12:15:35 S4 env[29693]:     at async Object.apiResolver (/vms/umami/node_modules/next/dist/server/api-utils/node.js:179:9)
Jul 22 12:15:35 S4 env[29693]:     at async NextNodeServer.runApi (/vms/umami/node_modules/next/dist/server/next-server.js:381:9)
Jul 22 12:15:35 S4 env[29693]:     at async Object.fn (/vms/umami/node_modules/next/dist/server/base-server.js:497:37)
Jul 22 12:15:35 S4 env[29693]:     at async Router.execute (/vms/umami/node_modules/next/dist/server/router.js:213:36)
Jul 22 12:15:35 S4 env[29693]:     at async NextNodeServer.run (/vms/umami/node_modules/next/dist/server/base-server.js:616:29) {
Jul 22 12:15:35 S4 env[29693]:   code: 'P2010',
Jul 22 12:15:35 S4 env[29693]:   clientVersion: '4.1.0',
Jul 22 12:15:35 S4 env[29693]:   meta: {
Jul 22 12:15:35 S4 env[29693]:     code: '1582',
Jul 22 12:15:35 S4 env[29693]:     message: "Incorrect parameter count in the call to native function 'convert_tz'"
Jul 22 12:15:35 S4 env[29693]:   }
Jul 22 12:15:35 S4 env[29693]: }
```

**Comment by hans362** [comment]
The issue seems to be resolved in the latest dev. Now the charts are correct.
![](https://user-images.githubusercontent.com/32535810/180363074-9517780d-b179-4508-9771-c229130f6665.png)

**Comment by mikecao** [comment]
Should be fixed in v1.35.0 now.

**Comment by hans362** [comment]
Yes, it's working fine now. Thanks!
